RIOT-OS/applications
# [WIP] RIOT native LwM2M client

This PR is a WIP to provide a basic RIOT native LwM2M client based on gcoap. Presently it includes the ability to register with a device management server as well as a SAUL based information reporting module for a temperature sensor, similar to #66.

RIOT presently imports the Wakaama client library as a package, and includes a similar example. However, unlike Wakaama the implementation in this PR will not include dynamic memory management. Also, initially this implementation will focus on providing sensor readings as simply as possible.

### Testing procedure
See [README.md](https://github.com/kb2ma/riot-apps/blob/wakaama-temp/wakaama-temp/README.md) to set up and run the app. I also have a rudimentary pytest repository, [lwm2m-pytest](https://github.com/kb2ma/lwm2m-pytest).

For temperature readings, I use a Microchip MCP9808 sensor with the RIOT JC42 driver. Adafruit [makes it easy](https://www.adafruit.com/product/1782) for those of us who live in the higher levels of the stack. :-)
